src/main/java/org/tikv/common/apiversion/CodecUtils.java

```diff
@@ -0,0 +1,18 @@
+package org.tikv.common.apiversion;
+
+import com.google.protobuf.ByteString;
+import org.tikv.common.codec.Codec.BytesCodec;
+import org.tikv.common.codec.CodecDataInput;
+import org.tikv.common.codec.CodecDataOutput;
+
+public class CodecUtils {
+  public static ByteString encode(ByteString key) {
+    CodecDataOutput cdo = new CodecDataOutput();
+    BytesCodec.writeBytes(cdo, key.toByteArray());
```

Comment:
we can have a fast path to take internal array out without copying when implementation of ByteString is actually ByteString.LiteralByteString or ByteString.BoundedByteString. 

To use package level protected method or classes, you can either create a utility class in our source tree within com.google.protobuf package or use reflection to make them accessible.